builtin/providers/aws/resource_aws_devicefarm_project.go

```diff
@@ -0,0 +1,111 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/devicefarm"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDeviceFarmProject() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDeviceFarmProjectCreate,
+		Read:   resourceAwsDeviceFarmProjectRead,
+		Update: resourceAwsDeviceFarmProjectUpdate,
+		Delete: resourceAwsDeviceFarmProjectDelete,
+
+		Schema: map[string]*schema.Schema{
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDeviceFarmProjectCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).devicefarmconn
+	region := meta.(*AWSClient).region
+
+	//	We need to ensure that DeviceFarm is only being run against us-west-2
+	//	As this is the only place that AWS currently supports it
+	if region != "us-west-2" {
+		return fmt.Errorf("DeviceFarm can only be used with us-west-2. You are trying to use it on %s", region)
```

Comment:
If we were to skip this test and the user set it to some other region, would the AWS API fail or would it do something unwanted?

If it'll fail anyway, maybe it's better to just skip this check and let the API call fail below, and then Terraform won't need to be updated if this service gets extended to work in other regions.